presto-main/src/main/java/com/facebook/presto/operator/PagesIndex.java

```diff
@@ -338,12 +338,13 @@ public LookupSource createLookupSource(List<Integer> joinChannels, Optional<Inte
             //        OUTER joins into NestedLoopsJoin and remove "type == INNER" condition in LocalExecutionPlanner.visitJoin()
 
             try {
-                LookupSourceFactory lookupSourceFactory = joinCompiler.compileLookupSourceFactory(types, joinChannels);
+                LookupSourceFactory lookupSourceFactory = joinCompiler.compileLookupSourceFactory(types, joinChannels, filterFunction.map(f -> f.getClass()));
```

Comment:
Cool it actually works. I used lambda because intelliJ complained here with "type mismatch" error. But javac compiles it correctly.